pkg/boot/grub/grub.go

```diff
@@ -126,6 +133,29 @@ func grubScanBLSEntries(mountPool *mount.Pool, variables map[string]string) ([]b
 	return images, nil
 }
 
+// Find and return the value of the key from a grubenv file
+func findkeywordGrubEnv(file string, fsRoot string, key string) (string, error) {
+	FileGrubenv, err := filepath.Glob(filepath.Join(fsRoot, file))
+	if FileGrubenv == nil || err != nil {
+		return "", err
+	}
+	relNamesFile, err := os.Open(FileGrubenv[0])
+	if err != nil {
+		return "", fmt.Errorf("[grubenv] Could not Open %s: %v", FileGrubenv[0], err)
+	}
+	grubenv, err := uio.ReadAll(relNamesFile)
+	if err != nil {
+		return "", fmt.Errorf("[grubenv] Could not Read %s: %v", FileGrubenv[0], err)
```

Comment:
I'm sorry, I forgot to mention this, but if you can use %w instead of %v, it will make life easier for future test writers :-)

Reply:
also, note, you do not need to print the file name. it is included in the error.

Reply:
Done. Thank you~